app/models/notification.rb

```diff
@@ -68,7 +68,7 @@ def set_notification_recipients
 
   def set_custom_attributes
     return unless notification_blueprint # let validation catch this.
-    self.actions = UINotifications::URLResolver.new(
+    self.actions ||= UINotifications::URLResolver.new(
```

Comment:
not sure  i follow, related to the comment blue on message validation on the blueprint.

Reply:
TBH I'm still not sure i follow this change - I'm still concerned that this is a "hack".

Reply:
the condition is `if notification_blueprint.actions.any?` which means it would be always overridden if blueprint defines some default actions

Reply:
@ares I understand the condition works, my concern is with how the actions is be used, I'm just wondering if this is inline with the original design of the notification actions (being part of a blueprint). I'm more than ok to review the notification implementation and to see if that makes sense or not, but was concerned with adding this without a good understanding on the impact.

Reply:
Ok, so what was the idea behind blueprint and actions? from the code I understood that blueprint is a notification template but you can override specific attributes. Attribute message already had `||=` therefore it felt like a bug that actions behave differently.

Reply:
@ohadlevy I know you're busy, but is there a chance you can take a look at this? I don't want it to slip off from my radar but the older it gets the bigger chance it will.